src/AWS.Deploy.CLI/Commands/TypeHints/FilePathCommand.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.Recipes;
+using AWS.Deploy.Common.Recipes.Validation;
+using AWS.Deploy.Common.TypeHintData;
+
+namespace AWS.Deploy.CLI.Commands.TypeHints
+{
+    /// <summary>
+    /// Typehint that lets the user specify an absolute path to a file.
+    /// </summary>
+    public class FilePathCommand : ITypeHintCommand
+    {
+        private readonly IConsoleUtilities _consoleUtilities;
+
+        public FilePathCommand(IConsoleUtilities consoleUtilities)
+        {
+            _consoleUtilities = consoleUtilities;
+        }
+
+        /// <summary>
+        /// Not implemented, specific files are not suggested to the user
+        /// </summary>
+        /// <returns>Empty list</returns>
+        public Task<List<TypeHintResource>?> GetResources(Recommendation recommendation, OptionSettingItem optionSetting) => Task.FromResult<List<TypeHintResource>?>(null);
+
+        /// <summary>
+        /// Prompts the user to enter a path to a file
+        /// </summary>
+        public Task<object> Execute(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            var userFilePath = _consoleUtilities
+               .AskUserForValue(
+                   string.Empty,
+                   recommendation.GetOptionSettingValue<string>(optionSetting),
+                   allowEmpty: true,
+                   resetValue: recommendation.GetOptionSettingDefaultValue<string>(optionSetting) ?? "");
+
+            return Task.FromResult<object>(userFilePath);
```

Comment:
I think we should validate the filepath points to an existing file. If it does not tell the user and prompt again. Assuming filepath is not empty.

Reply:
We do, it's done via an `IOptionSettingItemValidator` `FileValidator.cs`.

```
Dockerfile Path:
Specifies the absolute path to the Dockerfile as an alternative to the default location.
Enter value (default C:\Users\shovlia\source\repos\ASPWebAPI\ASPWebAPI\Dockerfile. Type <reset> to reset.):
asdf
The specified file asdf does not exist

Dockerfile Path:
Specifies the absolute path to the Dockerfile as an alternative to the default location.
Enter value (default C:\Users\shovlia\source\repos\ASPWebAPI\ASPWebAPI\Dockerfile. Type <reset> to reset.):
```

Reply:
At the time I didn't realize that `AskUserForValue` has a `validators` parameter.

`FileExistsValidator : `IOptionSettingItemValidator` implements the file check. I've wrapped that in `ValidateFileExists` here to match the signature for `AskUserForValue`